### PR TITLE
WMTS BoundingBox fix and WMTS template adjustment

### PR DESCRIPTION
**Pull request by AndrewAnnex**
fixes a bug where the WMTS BoundingBox coordinates flipped incorrectly if boundingbox element is used, as the coords must match the CRS order

updates WMTS XML template to avoid placing extra new lines in various places

### CODE REVIEW

File: src/titiler/core/titiler/core/templates/wmts.xml [hunk 1]
```diff
@@ -51,13 +51,13 @@
             </TileMatrixSetLink>
             <ResourceURL format="{{ media_type }}" resourceType="tile" template="{{ layer.tiles_url }}?{{ layer.query_string | escape }}" />
         </Layer>
-    {% endfor %}
+    {%- endfor %}
```
Comment (vincentsarago) [inline src/titiler/core/titiler/core/templates/wmts.xml:54]: why do we need `-` ?

**Comment by vincentsarago** [comment]
thanks @AndrewAnnex 

could you please add a note in https://github.com/developmentseed/titiler/blob/main/CHANGES.md#unreleased

File: src/titiler/core/titiler/core/templates/wmts.xml (hunk 1 above)
Comment (AndrewAnnex) [reply src/titiler/core/titiler/core/templates/wmts.xml:54]: it removes extra white space that's being inserted https://jinja.palletsprojects.com/en/stable/templates/#whitespace-control

**Comment by AndrewAnnex** [comment]
hmm I don't know why the black formatting check is failing, last commit I ran it locally and I just did again via the pre-commit (which I keep forgetting about installing on new systems I'm deving on)

**Pull request merged**